src/components/notebook-utils.js

```diff
@@ -58,6 +68,9 @@ export const notebookNameInput = ({ inputProps, ...props }) => h(ValidatedInput,
   }
 })
 
+export const getAnalysisType = fileName => Utils.switchCase(getExtension(fileName), ['ipynb', () => 'python'], ['rmd', () => 'r'])
```

Comment:
Hmm... I could see it either way. I chose the analysis type to be the language the analysis is in. Don't feel strongly either way

Reply:
Then I'd suggest we have a mapping from `file type` to `application` because
1. `ipynb` can map to both `python` and `r`
2. There may be apps in the near future where language is N/A (e.g. visualization tools such as Genome Browser)

Reply:
Also a TOL: For the reason #2 above, I feel using the terms `application` and `artifact` instead of `analysis` are more future-proof.